Flaky shared_stream test times out in CI

`airflow-core/tests/unit/triggers/test_shared_stream.py::test_ack_mode_queue_full_during_fanout_does_not_break_iteration` is timing out in CI.

I hit this in PR #68479, but it does not appear related to that PR. The same test is also failing on `main` with the same `TimeoutError`.

**Main failure example:**
https://github.com/apache/airflow/actions/runs/27607202688/job/81625031694

**Relevant failure:**

```text
airflow-core/tests/unit/triggers/test_shared_stream.py:1437: in test_ack_mode_queue_full_during_fanout_does_not_break_iteration
    await asyncio.wait_for(collect_full(), timeout=2.0)
...
E   TimeoutError
```

This seems to be a flaky or recently introduced timing issue in the shared stream trigger tests and is currently blocking unrelated PRs when the full core test suite runs.

--- a/airflow-core/tests/unit/triggers/test_shared_stream.py
+++ b/airflow-core/tests/unit/triggers/test_shared_stream.py
@@ -1434,6 +1434,15 @@ async def test_ack_mode_queue_full_during_fanout_does_not_break_iteration():
 
         # t_ok: drive its ack stream and collect the real event.
         async with _consume_in_background(t_ok.filter_shared_stream(s_ok)) as ok_collector:
+            # Let the poll loop fan the event out and force QueueFull on trigger 1's
+            # pre-filled queue *before* draining s_full. Otherwise collect_full can pull
+            # the filler off the queue first, the fan-out put then succeeds, and the
+            # overflow path is never taken.
+            deadline = asyncio.get_event_loop().time() + 1.0
+            while 1 not in group._failed_subscribers:
+                assert asyncio.get_event_loop().time() < deadline, "fan-out did not overflow trigger 1"
+                await asyncio.sleep(0)
+
             await asyncio.wait_for(collect_full(), timeout=2.0)
             ok_result = await ok_collector.wait_for(1)
 

